### PR TITLE
Lot of changes

### PR DESCRIPTION
**Pull request by Codestech1**
This pr will have improvements in translations etc.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,8 +12,7 @@ gradle-app.setting
 .gradletasknamecache
 
 # # Work around https://youtrack.jetbrains.com/issue/IDEA-116898
-# gradle/wrapper/gradle-wrapper.properties
-
+gradle/wrapper/gradle-wrapper.properties
```
Comment (GreatWyrm) [inline .gitignore]: Please re-comment this out, we want to keep the gradle-wrapper.properties file

File: build.gradle.kts [hunk 2]
```diff
@@ -11,12 +12,20 @@ repositories {
 }
 
 dependencies {
-    compileOnly("com.github.Minestom:Minestom:809d9516b2")
-    compileOnly("com.github.CityWideMC:CityStom:8f9625559f")
+    compileOnly("com.github.CityWideMC.Minestom:Minestom:fe815c61d6")
```
Comment (GreatWyrm) [inline build.gradle.kts]: Why are we switching away from core Minestom? This needs a good explanation.

File: src/main/java/com/arcanewarrior/disguises/commands/HideCommand.java [hunk 3]
```diff
@@ -21,16 +16,5 @@ public HideCommand(DisguiseManager disguiseManager) {
                 suggestion.addEntry(new SuggestionEntry(string));
             }
         });
```
Comment (GreatWyrm) [inline src/main/java/com/arcanewarrior/disguises/commands/HideCommand.java]: Why was this deleted?

File: src/main/java/com/arcanewarrior/disguises/Disguise.java [hunk 4]
```diff
@@ -14,6 +15,11 @@ public Disguise(@NotNull EntityType entityType, @NotNull Player player, @NotNull
         super(entityType);
         this.player = player;
         this.manager = manager;
+        if(MinestomDisguises.getInstance().getConfiguration().getTable("disguises").getBoolean("translate-player-names", true)) {
```
Comment (GreatWyrm) [inline src/main/java/com/arcanewarrior/disguises/Disguise.java]: This is not really that nice, it should be a config that's held onto by disguise manager, not accessed statically

File: src/main/java/com/arcanewarrior/disguises/commands/ShowCommand.java [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package com.arcanewarrior.disguises.commands;
 
 import com.arcanewarrior.disguises.DisguiseManager;
+import net.minestom.server.MinecraftServer;
```
Comment (GreatWyrm) [inline src/main/java/com/arcanewarrior/disguises/commands/ShowCommand.java]: Random import? Please clean this up.

File: src/main/java/com/arcanewarrior/disguises/listeners/translation/SpawnTranslation.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.arcanewarrior.disguises.listeners.translation;
+
+import com.arcanewarrior.disguises.DisguiseManager;
+import net.minestom.server.event.player.PlayerPacketOutEvent;
+
+public final class SpawnTranslation {
+    public static void listener(PlayerPacketOutEvent event, DisguiseManager parentManager) {
+        /*if (event.getPacket() instanceof SpawnPlayerPacket packet) {
```
Comment (GreatWyrm) [inline src/main/java/com/arcanewarrior/disguises/listeners/translation/SpawnTranslation.java]: I'd recommend removal of this class if you have it entirely commented out.

**Review by GreatWyrm — CHANGES_REQUESTED** [review]
Appreciate the further help!

However, some of your changes need an explanation, and your commit messages generally being "changes" and variations thereof aren't clarifying why these changes have been made.

**Comment by Codestech1** [comment]
I did changes, but i forgot about few things, one with them needs merge pr on minestom, so i'd converted it to draft again.

File: .gitignore (hunk 1 above)
Comment (GreatWyrm) [inline .gitignore]: Okay, I asked for this to be commented back out, and while the deleted file is back, this is still not commented out.

File: src/main/java/com/arcanewarrior/disguises/listeners/DisguiseEvents.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        node.addListener(EntityPotionRemoveEvent.class, event -> EffectTranslation.listener(event, parentManager));
+        return node;
+    }
+
+    @SuppressWarnings("UnstableApiUsage")
+    private EventNode<EntityEvent> miscTranslations() {
+        EventNode<EntityEvent> node = EventNode.type("misc", EventFilter.ENTITY);
+        node.addListener(PlayerPacketEvent.class, this::delegatePacket);
+        node.addListener(PlayerPacketOutEvent.class, this::delegatePacket);
+        node.addListener(PlayerMoveEvent.class, event -> MoveTranslation.listener(event, parentManager));
```
Comment (GreatWyrm) [inline src/main/java/com/arcanewarrior/disguises/listeners/DisguiseEvents.java:62]: Why was the move packet translation changed to a move event instead of working with the packets?

File: src/main/java/com/arcanewarrior/disguises/listeners/translation/InventoryTranslation.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import net.minestom.server.event.player.PlayerSwapItemEvent;
+import net.minestom.server.event.trait.PlayerEvent;
+
+public final class InventoryTranslation {
+    public static void listener(PlayerEvent event, DisguiseManager parentManager) {
+        if(event instanceof PlayerChangeHeldSlotEvent e) {
+            Player player = e.getPlayer();
+            Disguise disguise = parentManager.getPlayerDisguise(player);
+            if (disguise != null)
+                MinecraftServer.getSchedulerManager().scheduleNextTick(() -> disguise.setItemInMainHand(player.getItemInMainHand()));
```
Comment (GreatWyrm) [inline src/main/java/com/arcanewarrior/disguises/listeners/translation/InventoryTranslation.java:19]: Why is this a scheduleNextTick?

File: src/main/java/com/arcanewarrior/disguises/listeners/translation/InventoryTranslation.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+            Player player = e.getPlayer();
+            Disguise disguise = parentManager.getPlayerDisguise(player);
+            if(disguise != null)
+                for(EquipmentSlot slot : EquipmentSlot.values())
+                    disguise.setEquipment(slot, player.getEquipment(slot));
+        } else if(event instanceof PlayerSwapItemEvent e) {
+            Player player = e.getPlayer();
+            Disguise disguise = parentManager.getPlayerDisguise(player);
+            if(disguise != null) {
+                player.setItemInMainHand(e.getMainHandItem());
```
Comment (GreatWyrm) [inline src/main/java/com/arcanewarrior/disguises/listeners/translation/InventoryTranslation.java]: Should be disguise.setItemInMain/Off hand

File: src/main/java/com/arcanewarrior/disguises/listeners/translation/MoveTranslation.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import com.arcanewarrior.disguises.DisguiseManager;
+import net.minestom.server.coordinate.Pos;
+import net.minestom.server.entity.Player;
+import net.minestom.server.event.player.PlayerMoveEvent;
+
+public final class MoveTranslation {
+    public static void listener(PlayerMoveEvent event, DisguiseManager parentManager) {
+        Player player = event.getPlayer();
+        Disguise disguise = parentManager.getPlayerDisguise(player);
+        if (disguise != null) {
```
Comment (GreatWyrm) [inline src/main/java/com/arcanewarrior/disguises/listeners/translation/MoveTranslation.java:13]: I'm not fond of this move translation, as teleporting the entity is a lot less smooth than translating the move packets from player to entity.

**Review by GreatWyrm — CHANGES_REQUESTED** [review]
Left some comments on areas I think need improvement

Like I stated before, you need to explain why you are doing things. Commit messages with "Changes" are not helpful to me to understand why these changes were made or if I think there could be a better way to go about what you want to do.

I do thank you for the contributions you have made, but please clarify more about why you're making these changes.

File: src/main/java/com/arcanewarrior/disguises/listeners/translation/MoveTranslation.java (hunk 10 above)
Comment (Codestech1) [reply src/main/java/com/arcanewarrior/disguises/listeners/translation/MoveTranslation.java:13]: Minestom use teleports in their move listener and it's fine. I tested it btw and it's smooth.

File: src/main/java/com/arcanewarrior/disguises/listeners/translation/InventoryTranslation.java (hunk 8 above)
Comment (Codestech1) [reply src/main/java/com/arcanewarrior/disguises/listeners/translation/InventoryTranslation.java:19]: because without this it doesn't work

File: src/main/java/com/arcanewarrior/disguises/listeners/DisguiseEvents.java (hunk 7 above)
Comment (Codestech1) [reply src/main/java/com/arcanewarrior/disguises/listeners/DisguiseEvents.java:62]: Because if we will work in packets, cancelled movement (in events) won't be cancelled in disguise.
